### ISSUE TITLE
Requests to /api/user always fails due to CSRF token

### ISSUE THREAD

**Issue by FinucaneDesign** [laravel/passport#255]
All api calls fail with 401 Unauthenticated. 

I can make them work by commenting out the vaildCsrf check.  I am setting the X-CSRF-TOKEN with axios using a token created by php function csrf_token() in the master layout.  It looks like the csrf token you get from csrf_token() does not match the csrf token passport is including in the laravel_token cookie.  The passport csrf doesn't seem to update the csrf on login.

From Veiws/Layout/app.blade.php
```
    <script>
        window.Laravel = <?php echo json_encode([
            'csrfToken' => csrf_token(),
        ]); ?>
    </script>
```

From Bootstrap.js
`    axios.defaults.headers.common['X-CSRF-TOKEN'] = Laravel.csrfToken;`

Edit Passport TokenGuard.php to make request work. 
```
    if (! $this->validCsrf($token, $request) ||
        time() >= $token['expiry']) {
        # return;
        Log::info('token error', [$token['csrf'], $request->header('X-CSRF-TOKEN'), hash_equals(
                $token['csrf'], (string) $request->header('X-CSRF-TOKEN'))]);
    }
```

Results:
`local.INFO: token error ["elW55K775dGSM9HkGmjtyE8FAZoCiAwJHMkICUYx","WLfyjiyjw7iZ2Mh8cvUR1DYg2bniYdWmuNX7Jp5W",false] `

Then logout and login:
```
local.INFO: token error ["elW55K775dGSM9HkGmjtyE8FAZoCiAwJHMkICUYx","Zi9jYn8yDbYFwv4PmK3eaByYjLZza3gpeU0wmWWn",false] 
local.INFO: token error ["elW55K775dGSM9HkGmjtyE8FAZoCiAwJHMkICUYx","ozxDRnxGtQ4p4Fs7wWFxldQe9tsRFuWren6J8HpA",false] 
```

Request headers:
```
Accept:application/json, text/plain, */*
Accept-Encoding:gzip, deflate, sdch
Accept-Language:en-US,en;q=0.8
Cache-Control:no-cache
Connection:keep-alive
Cookie:laravel_token=eyJpdiI6IjRsTjVlYUhFdzBjUTBjZFNRRTRWc3c9PSIsInZhbHVlIjoiRks3R1BzUXhzWnNZUVRSREM2NGxpZ2pSVFlPNFdjQVFEdk5XUFFhc1VyM1RvM1wvSDBPVWJyKzQxc3hwd050d0xpVU5hck9SaFh0RkFocEVERVY1NE9vTE5FQ0FwSExvSHZcLzFBalBJVTNQUlpjK1IrdU5pbzdKVTFNeU8yMXFVUG12cDhGMnZEWHBxbThxM2dFOGVBSG5yZjdHQ21wTGN6dHhIQk5JcDFZTlpLZCt3SHhxS29JTytBRk9vT3BKaTJjQ01xa29HNGJTVWsyQXVsYUhySTFvcHdjUG9MNzVvUU1ydWx0cVVtOWpmUmZBaXduM0xrRjllcjhOMUw3d1BOQVFha05xb0Z4c0pIWVIyb09sb3A0dz09IiwibWFjIjoiNTBjNGMyNTg1ZTdmYmM4YmU3ZDhlODBhYjNhNzA0Y2E1ZWZmNzE4NmQ2NjhiYWFjZTJlZjExNjM4NTJlMGQ1OCJ9; XSRF-TOKEN=eyJpdiI6IkQ2eU5XZVZFNGlJQlAxdWl3SGxTU2c9PSIsInZhbHVlIjoibWV0XC9UMlZLOUpyZXp2MFZrXC8yem1nNTlMMk1FMHlTekQyU0xDSWFZUGZmZXdtbU04aENYZHY1YTd2WmQ2VkYxRFhLN2lTUWozZ2Y4dlBKc1ROSUhTZz09IiwibWFjIjoiN2IzZGRmN2MwNmYzZjM3YzFmM2I4MzlmMTM4MzZlMDBjYTgyYzY1MTk5N2VlZmZkMjIzYzcwYjQ2YmUxOGU2OSJ9; laravel_session=eyJpdiI6InNWUUdhQWRBN01VeUg1UlwvSk1zRllRPT0iLCJ2YWx1ZSI6InA4YmY2djM0TEZKSnhiY04xZWlTQ3RTXC9yVEgzaTVJNDBPZWZSOFFFYTBiMU5iT1FOVlNXOXREa3dnamp4NTdSUTNsRUVtM0MwWGphdkFzQmxiUkpWdz09IiwibWFjIjoiNWE4NWY3N2QxNjE4YjkyMzI5ZGI4MWI0MmI3Zjk1MjM4NTFiOGQyMjNmZThkYTExMDAyNDFhMTZmZDhjZDdlZCJ9; laravel_token=eyJpdiI6IndUcVREblB2VnRNOEJmeDE3VzVSNmc9PSIsInZhbHVlIjoiTzFSYkVVUzVQTUtMZXdqd0M1b1dvRjg5NlRFR0l5OG44c010XC9NZlkzQm11YSt0bDdzUzRBTkFicWh4RjJTNEFSOHdjaDl4dFVlaHdoQVdUSVU4SkExMXBiMmlxUUVVTENUc3VwVDdybzdmY1Q1R3IybEJwdlFKU2pzTkNlSlUxMmhHdFU1QVFQZWlkamk3Ritpbm5QNHozb3dVdk96ckZzUnNKTmFIajRpcnJWREVWNUswREZQeWdcL1BFRGJoYUFGejRuQVBOOWMxZllXUTM0dEJhXC9RN1FFd3U5YU1FWnlDTmsxbWJHbXZMSmdVYWdkQ3o3UU11RXVTUTJxTmZFOG45MWNDcjNEaVwvUURDTFAyU3kwQ3V3PT0iLCJtYWMiOiJmNTU5YzEzYmRmYTRjZGFkNGNlMWI2ZDFjZDlmNjJlMTlmYzUzOTVmN2ZmYTQzMTUxZjIzM2FlODdiM2UyNmZlIn0%3D; XSRF-TOKEN=eyJpdiI6Ikx6NlkyZlBKckdtWXlIbFRMNWpqeGc9PSIsInZhbHVlIjoiWFh0V1FXQUFMdUt6ZTVzK3NyeWdrV0w0V1RsN0gxbnVcL3VLUHVuNE5XQzlFOEkwT1VZN001NTJyTUREOTJTek5jMGtKbVNHNzBIZWpMeFFld3R0SnVBPT0iLCJtYWMiOiI2MWQxODk1MzI0N2NiZWU5YTg4ZWVjYjVjY2EyZmYxNzNlM2JlNTRmZWY4YTdjNWE2NzE0ZjEyNDZjMjQ3ZjA5In0%3D; laravel_session=eyJpdiI6ImM4XC9FZVI0S2tacDFFSE9NaTRQODhBPT0iLCJ2YWx1ZSI6Ik50Sjg0dDB3YWVDSmpOZDZWNUZJajlMQ0JtNzZNNERaeUVaK1VhN0ZNcWVseFRBbVd1N2hsNlErMlwvdUNSY0NkZDRUclh2aXpVeXRMc0pFWmx5NURVZz09IiwibWFjIjoiNDY2YTc5NGQ4Yzk3NTc3N2JlZGE1YjY1MGIxYTc3OTlkYTFlMDk4MWI2YmQzMzY5Y2IxODFjYTliNmNlMDRkZCJ9
DNT:1
Host:web.dev
Pragma:no-cache
Referer:http://web.dev/dashboard
User-Agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36
X-CSRF-TOKEN:WLfyjiyjw7iZ2Mh8cvUR1DYg2bniYdWmuNX7Jp5W
X-XSRF-TOKEN:eyJpdiI6Ikx6NlkyZlBKckdtWXlIbFRMNWpqeGc9PSIsInZhbHVlIjoiWFh0V1FXQUFMdUt6ZTVzK3NyeWdrV0w0V1RsN0gxbnVcL3VLUHVuNE5XQzlFOEkwT1VZN001NTJyTUREOTJTek5jMGtKbVNHNzBIZWpMeFFld3R0SnVBPT0iLCJtYWMiOiI2MWQxODk1MzI0N2NiZWU5YTg4ZWVjYjVjY2EyZmYxNzNlM2JlNTRmZWY4YTdjNWE2NzE0ZjEyNDZjMjQ3ZjA5In0=
```

**Comment by FinucaneDesign** [comment]
Looks like it was a bug in chrome.  After I deleted the cookies in dev_tools, chrome was still attaching the deleted cookie wtf!?.  No problem in firefox or safari, or incognito window of chrome.  I when to delete cookies from my dev site and laravel_token was in the list twice.  Cleared cookies and it working as expected.

**Comment by shawnfromportland** [comment]
For others still encountering this, I was able to resolve this error that dozens or hundreds of people report online, but none of the existing solutions worked for me, except for the following.  
using  Laravel 5.8, following docs to set up Passport.
  
In my case, although I was including the csrf token in a meta tag, it was not being picked up as the Passport documentation states that laravel will by default. from the docs (https://laravel.com/docs/5.8/passport#consuming-your-api-with-javascript):
>" the default Laravel JavaScript scaffolding instructs Axios to always send the X-CSRF-TOKEN and  X-Requested-With headers. "  
  
and then the example is provided:
```
// In your application layout...
<meta name="csrf-token" content="{{ csrf_token() }}">

// Laravel's JavaScript scaffolding...
window.axios.defaults.headers.common = {
    'X-Requested-With': 'XMLHttpRequest',
};
```

Well that would be great if it worked that way by default as the docs state, but in my case I had to explicitly set the axios default to include the contents of said csrf-token meta tag before making the axios request. like this:
```
window.axios.defaults.headers.common = {
						'X-Requested-With': 'XMLHttpRequest',
						'X-CSRF-TOKEN' : document.querySelector('meta[name="csrf-token"]').getAttribute('content')
				};
```  

In my case, this was the only thing that allowed me to get past the 401 unauthorized error, which seems to indicate either:  
A) something in my project's configuration has altered the default behavior of axios requests by not setting them automatically include the csrf-token based on the meta tag if present or  
 B) the docs are inaccurate on this one point, and in the present version of Laravel and Passport that is not the default behavior as stated.
